src/calendar-backend/Calendar.Api/Controllers/CalendarController.cs

```diff
@@ -0,0 +1,162 @@
+    private readonly ILogger<CalendarController> logger;
+
+    public CalendarController(ICalendarService calendarService, IEventService eventService, IMapper mapper, ILogger<CalendarController> logger)
+    {
+        this.calendarService = calendarService;
+        this.eventService = eventService;
+        this.mapper = mapper;
+        this.logger = logger;
+    }
+
+    #region UserCalendar
+
+    [HttpPost]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<UserCalendarDTO>> AddCalendar([FromBody] CreateUserCalendarDTO calendar)
+    {
+        if (calendar == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await calendarService.AddCalendarAsync(mapper.Map<UserCalendar>(calendar));
+
+        return Ok(mapper.Map<UserCalendarDTO>(result));
+    }
+
+    [HttpGet]
+    //[Authorize(Roles = "calendar-viewer,calendar-editor")]
+    [Authorize]
+    public async Task<ActionResult<IEnumerable<UserCalendarDTO>>> GetCalendarsByNames()
+    {
+        var groups = base.HttpContext.User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.GroupSid);
+        var testdata = new List<string>
+        {
+            "TINF21AI"
+        }; // TODO: use groups from claim 
+        var calendar = await calendarService.GetCalendarsByNamesAsync(testdata);
+
+        return Ok(mapper.Map<IEnumerable<UserCalendarDTO>>(calendar));
+    }
+
+    [HttpGet("{calendarId}")]
+    // [Authorize(Roles = "viewer,editor")]
+    public async Task<ActionResult<UserCalendarDTO>> GetCalendarById(string calendarId, [FromQuery] bool includeEvents = false)
+    {
+        var calendar = await calendarService.GetCalendarByIdAsync(calendarId, includeEvents);
+        if (calendar == null)
+            return NotFound();
+        return Ok(mapper.Map<UserCalendarDTO>(calendar));
+    }
+
+    [HttpPut("{calendarId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<UserCalendarDTO>> EditCalendar(string calendarId, [FromBody] UpdateUserCalendarDTO calendar)
+    {
+        if (calendar == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await calendarService.UpdateCalendarAsync(calendarId, mapper.Map<UserCalendar>(calendar));
+        if (result == null)
+            return NotFound();
+        return Ok(mapper.Map<UserCalendarDTO>(result));
+    }
+
+
+    [HttpDelete("{calendarId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<bool>> DeleteCalendar(string calendarId)
+    {
+        var success = await calendarService.DeleteCalendarByIdAsync(calendarId);
+        return Ok(success);
+    }
+
+    #endregion
+
+    #region CalendarEvents
+
+    [HttpPost("{calendarId}/event")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<CalendarEventDTO>> AddEvent(string calendarId, [FromBody] CreateCalendarEventDTO calendarEvent)
+    {
+        if (calendarEvent == null)
+        {
```

Comment:
Kann keine Lecture ID angeben
![image](https://github.com/mhummel-artiso/lecture-calendar/assets/97592367/757545be-b6f5-4811-b5dc-9914d70f1cd5)

Evtl. DTO anpassen

Reply:
Der user soll keine lecture id angeben, ich generiere jetzt in der API eine id

Reply:
Und wie soll man dann beim Event erstellen eine Lecture verknüpfen können?

Reply:
ich meinte event id

Reply:
TODO: fill event wiht lecture data